File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let sum = brown + yellow;
    
    for (let width = 3; width <= brown; width++) {
        if (sum % width === 0) {
            const height = sum / width;
            if ((height - 2) * (width - 2) === yellow) {
                return [height, width]
            }
        }
    }
    
}

File: 프로그래머스/2/42842. 카펫/카펫.py


def solution(brown, yellow):
    allTiles = brown + yellow
    
    for x in range(3, int(allTiles ** 0.5) + 1):
        if allTiles % x == 0:
            y = allTiles // x
            if (x - 2) * (y - 2) == yellow:
                return sorted([x, y], reverse=True)